docker/docker.go

```diff
@@ -150,8 +154,41 @@ func localImageExists(ctx context.Context, dClient *client.Client, hostname stri
 	return true, nil
 }
 
-func pullImage(ctx context.Context, dClient *client.Client, hostname string, containerImage string) error {
-	out, err := dClient.ImagePull(ctx, containerImage, types.ImagePullOptions{})
+func pullImage(ctx context.Context, dClient *client.Client, hostname string, containerImage string, prs []v3.PrivateRegistry) error {
+
+	pullOptions := types.ImagePullOptions{}
+	prsMap := make(map[string]v3.PrivateRegistry)
+	// build the Private Registry map
+	if len(prs) > 0 {
+		for _, pr := range prs {
+			if pr.URL == "" {
+				pr.URL = "docker.io"
+			}
+			prsMap[pr.URL] = pr
+		}
+	}
+	// We do this if we have some docker.io login information
+	if strings.HasPrefix(containerImage, "docker.io") ||
+		!hasURL(containerImage) {
+		if pr, ok := prsMap["docker.io"]; ok {
+			regAuth, err := getRegistryAuth(pr)
+			if err != nil {
+				return err
+			}
+			pullOptions.RegistryAuth = regAuth
+		}
+	}
+	// We have a registry, but it's not docker.io
+	if hasURL(containerImage) &&
+		!strings.HasPrefix(containerImage, "docker.io") {
+		// this could be public or private, ImagePull() can handle it
+		// if we provide a PrivilegeFunc
+		regURL := strings.Split(containerImage, "/")[0]
```

Comment:
see comment below...there might be some standard library for parsing the docker image, lets find and use it